content/articles/wavelet-transform-analysis-of-1d-signals-using-python/index.md

```diff
@@ -0,0 +1,221 @@
+
+- Universal threshold(VishuShrink).
+- Bayes Shrink(this is the default method).
+
+By default  the `Denoise_wavelet()` function is used  for images. However, in this tutorial, we will use it for 1-D signals, meaning that the attributes related to the image are dropped. The syntax for using this function is:
+
+```python
+y = denoise_wavelet(x, wavelet='db1', mode='soft', wavelet_levels=n, method='BayesShrink', rescale_sigma='True')
+```
+
+Where:
+- `X`: Is the input noise signal.
+- `Y`: Is the denoised output.
+- `Wavelet`: This is the name of wavelet used.
+- `Mode`: It is the thresholding mode used. There is two thresholding mode, i.e. soft and hard threshold.
+- `Wave_levels`: This is the decomposition levels. Note that this must be an integer.
+- `Method`: It is the thresholding method used. Here we use any of the two supported methods, ie. VishuShrink and BayesShrink.
+- `Rescale-sigma`: There are only two options. These options are `True` or `false`. For 1-D signals, it should be set to `true`. It is to rescale the noise variance signal if it is internally rescaled.
+
+### Python code for denoising ecg signal
+First, we need to import some packages as shown below:
+
+```Python
+import numpy as np
+import PyWavelets as pywt
+from skimage.restoration import denoise_wavelet
+import matplotlib.pyplot as plt
+```
+
+- The `numpy` package handles mathematical and logical operations on arrays. 
+- The `pywt` package performs wavelet transform for the input signal. We then import the `denoise_wavelet()` function from the `skimage` package. 
+- Finally, for any plot in Python, the `matplotlib` package is used.
+
+Read the data from the in-built database. Python has an in-built ecg database. It makes it more efficient since we do not need to look for the data from an external source. To read this data, we use the code below:
+
+```Python
+x = pywt.data.ecg().astype(float)/256  # In-built ecg data is read
+```
+
+The signal obtained from the database is noise-free. We need to add noise to it to perform the denoising operation. To add the noise, we first define the noise variance. The noise variance is the noise energy per sample.
+
+```python
+sigma = 0.05  # Noise variance
+x_noisy = x + sigma * np.random.randn(x.size)   # Adding noise to signal
+```
+
+To add the noise to our signal(x), we use the formula `x + sigma * np.random.randn(x.size)`. 
+
+The `randn()` is used to add a random noise in consideration to the noise variance `sigma`.
+
+Let us now perform wavelet denoising using the `denoise_wavelet()` function using the syntax described earlier, as shown below:
+
+```python
+# Wavelet denoising
+x_denoise = denoise_wavelet(x_noisy, method='BayesShrink', mode='soft', wavelet_levels=3, wavelet='sym8', rescale_sigma='True')
+```
+
+We plot the noisy signal `x_noisy` and the denoised signal `x-denoise` for visualisation.
+
+```python
+plt.figure(figsize=(20, 10), dpi=100)
+plt.plot(x_noisy)
+plt.plot(x_denoise)
+```
```

Comment:
This code snippet won't show the image. You are missing the `plt.show()` snippet.